physics/test/ThermoIce0_test.cpp

```diff
@@ -0,0 +1,142 @@
+        }
+        std::string getName() const override { return "IceTemperatureData"; }
+
+        void setData(const ModelState::DataMap&) override
+        {
+            cice[0] = 0.99;
+            hice[0] = 0.001 / cice[0]; // Here we are using the true thicknesses
+            hice0[0] = hice[0];
+            hsnow[0] = 0.;
+            sss[0] = 32.;
+            sst[0] = Module::getImplementation<IFreezingPoint>()(sss[0]);
+            snow[0] = 0.;
+            tf[0] = Module::getImplementation<IFreezingPoint>()(sss[0]);
+            mlbhc[0] = 4.29151e7;
+            tice0[0] = -9.;
+            qio[0] = 0.;
+        }
+
+        HField hice;
+        HField hice0;
+        HField cice;
+        HField hsnow;
+        HField sst;
+        HField sss;
+        HField tf;
+        HField snow;
+        HField mlbhc; // Mixed layer bulk heat capacity
+        HField tice0;
+        HField qio;
+
+        ModelState getState() const override { return ModelState(); }
+        ModelState getState(const OutputLevel&) const override { return getState(); }
+    } atmoState;
+    atmoState.setData(ModelState::DataMap());
+
+    // Supply the atmospheric boundary arrays, without an entire
+    // IAtmosphereBoundary implementation
+    HField qow;
+    qow.resize();
+    ModelComponent::registerExternalSharedArray(ModelComponent::SharedArray::Q_OW, &qow);
+
+    HField qia;
+    qia.resize();
+    ModelComponent::registerExternalSharedArray(ModelComponent::SharedArray::Q_IA, &qia);
+
+    HField dqia_dt;
+    dqia_dt.resize();
+    ModelComponent::registerExternalSharedArray(ModelComponent::SharedArray::DQIA_DT, &dqia_dt);
+
+    HField qic;
+    qic.resize();
+    ModelComponent::registerExternalSharedArray(ModelComponent::SharedArray::Q_IC, &qic);
+
+    HField subl;
+    subl.resize();
+    ModelComponent::registerExternalSharedArray(ModelComponent::SharedArray::SUBLIM, &subl);
+
+    class FluxData : public IFluxCalculation {
+    public:
+        FluxData()
+            : IFluxCalculation()
+        {
+        }
+        std::string getName() const override { return "FluxData"; }
+
+        void setData(const ModelState::DataMap&) override
+        {
+            qow[0] = 0;
+            qia[0] = 0;
+            dqia_dt[0] = 0;
+            subl[0] = 0;
+        }
+
+        ModelState getState() const override { return ModelState(); }
+        ModelState getState(const OutputLevel&) const override { return getState(); }
+
+        void update(const TimestepTime&) override { }
+    } fluxData;
+
+    fluxData.setData(ModelState::DataMap());
+
+    TimestepTime tst = { TimePoint("2000-01-01T00:00:00"), Duration(600) };
+    ThermoIce0 ti0t;
+    ti0t.configure();
+    ti0t.setData(ModelState::DataMap());
+    ti0t.update(tst);
+
+    ModelArrayRef<ModelComponent::SharedArray::H_ICE, MARBackingStore> hice(ModelComponent::getSharedArray());
+    // So little ice should be reduced to zero
+    REQUIRE(hice[0] == 0.);
```

Comment:
e.g. my guess is this test is trying to establish that the height(?) of the ice is being reset or being constantly pinned to zero or matches a golden value (it all melts) - however it is impossible to be sure without a comment

I have no way of knowing if this is likely to fail sporadically (due to randomness) without inspecting the rest of the code. This kind of information would be useful in a comment at the top of the test

Reply:
Resolved by adding an explanatory comment.